qiskit/algorithms/gradients/finite_diff_estimator_gradient.py

```diff
@@ -0,0 +1,96 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Gradient of Sampler with Finite difference method."""
+
+from __future__ import annotations
+
+from typing import Sequence
+
+import numpy as np
+
+from qiskit.circuit import Parameter, QuantumCircuit
+from qiskit.opflow import PauliSumOp
+from qiskit.primitives import BaseEstimator
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+
+from .base_estimator_gradient import BaseEstimatorGradient
+from .estimator_gradient_result import EstimatorGradientResult
+
+
+class FiniteDiffEstimatorGradient(BaseEstimatorGradient):
+    """
+    Compute the gradients of the expectation values by finite difference method.
+    """
+
+    def __init__(self, estimator: BaseEstimator, epsilon: float = 1e-2, **run_options):
```

Comment:
Is there a reason for the default value of `epsilon`? I am wondering what is nice...

Reply:
It should be large enough to not be affected by noise in the objective function but as small as possible for best accuracy. For SV simulations we can choose this very small but with shots we theoretically need to choose something like ~1/\sqrt(shots). In practice I think people use e.g. 0.1 for 1000 shots or smaller for more shots

Reply:
I don't have a strong opinion. If people will use 1e-1 more in practice, 1e-1 sounds better.

Reply:
So the issue is whether the default should take sampling noise into account.

Reply:
Gradient framework: `1e-6`
https://github.com/Qiskit/qiskit-terra/blob/a7e87794ff687074122c7747585f4de133d5f17d/qiskit/opflow/gradients/circuit_gradients/param_shift.py#L49

ScipyOptimizer has scipy's default values.
https://github.com/Qiskit/qiskit-terra/blob/f65f08848fda5b301c83f4148dd2be4e8bd13eff/qiskit/algorithms/optimizers/scipy_optimizer.py#L129-L131

Perhaps, it might be nice **not** to have the default values. Users should set it explicitly.

Reply:
Appropriate epsilon value depends on number of shots and the number of shots is not always included in `run_options`. So, I agree on Hamamura-san's idea not to set the default value of epsilon.

Reply:
Since it's `1e-6` in the existing gradient framework, How about setting the default `epsilon` value to `1e-6`? When I hear an `epsilon` option, I feel it should have some default value. Users can change `epsilon` by themselves if they want to.

Reply:
There are definitely users who will tell us in Slack or other places that it doesn't work when they work with sampling noise. If we support them, we can have a default.